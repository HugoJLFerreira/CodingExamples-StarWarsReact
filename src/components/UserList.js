import React from "react";
import PropTypes from "prop-types";
import { styled, alpha } from "@mui/material/styles";
import Box from "@mui/material/Box";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TablePagination from "@mui/material/TablePagination";
import TableRow from "@mui/material/TableRow";
import TableSortLabel from "@mui/material/TableSortLabel";
import Toolbar from "@mui/material/Toolbar";
import Paper from "@mui/material/Paper";
import SearchIcon from "@mui/icons-material/Search";
import InputBase from "@mui/material/InputBase";
import { visuallyHidden } from "@mui/utils";
import PlanetDialog from "./PlanetDialog";

function descendingComparator(a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function getComparator(order, orderBy) {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

// This method is created for cross-browser compatibility, if you don't
// need to support IE11, you can use Array.prototype.sort() directly
function stableSort(array, comparator) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) {
      return order;
    }
    return a[1] - b[1];
  });
  return stabilizedThis.map((el) => el[0]);
}

const headCells = [
  {
    id: "name",
    numeric: false,
    disablePadding: true,
    label: "Name",
    width: "150px",
  },
  {
    id: "height",
    numeric: true,
    disablePadding: false,
    label: "Height (cm)",
  },
  {
    id: "mass",
    numeric: true,
    disablePadding: false,
    label: "Mass (kg)",
  },
  {
    id: "created",
    numeric: false,
    disablePadding: false,
    label: "Created",
  },
  {
    id: "edited",
    numeric: false,
    disablePadding: false,
    label: "Edited",
  },
  {
    id: "planet",
    numeric: false,
    disablePadding: false,
    label: "Planet",
    width: "100px",
  },
];

function EnhancedTableHead(props) {
  const { order, orderBy, onRequestSort } = props;
  const createSortHandler = (property) => (event) => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map((headCell) => (
          <TableCell
            key={headCell.id}
            padding="normal"
            sortDirection={orderBy === headCell.id ? order : false}
            className="Table-Row"
          >
            <TableSortLabel
              active={orderBy === headCell.id}
              direction={orderBy === headCell.id ? order : "asc"}
              onClick={createSortHandler(headCell.id)}
            >
              {headCell.label}
              {orderBy === headCell.id ? (
                <Box component="span" sx={visuallyHidden}>
                  {order === "desc" ? "sorted descending" : "sorted ascending"}
                </Box>
              ) : null}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

EnhancedTableHead.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.oneOf(["asc", "desc"]).isRequired,
  orderBy: PropTypes.string.isRequired,
};

const Search = styled("div")(({ theme }) => ({
  position: "relative",
  borderRadius: theme.shape.borderRadius,
  backgroundColor: alpha(theme.palette.common.white, 0.15),
  "&:hover": {
    backgroundColor: alpha(theme.palette.common.white, 0.25),
  },
  marginRight: theme.spacing(2),
  marginLeft: 0,
  width: "100%",
  [theme.breakpoints.up("sm")]: {
    marginLeft: theme.spacing(3),
    width: "auto",
  },
}));

const SearchIconWrapper = styled("div")(({ theme }) => ({
  padding: theme.spacing(0, 2),
  height: "100%",
  position: "absolute",
  pointerEvents: "none",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
}));

const StyledInputBase = styled(InputBase)(({ theme }) => ({
  color: "inherit",
  "& .MuiInputBase-input": {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: "20ch",
    },
  },
}));

const EnhancedTableToolbar = (props) => {
  const { handleFilter } = props;

  return (
    <Toolbar>
      <Search>
        <SearchIconWrapper>
          <SearchIcon />
        </SearchIconWrapper>
        <StyledInputBase
          placeholder="Searchâ€¦"
          inputProps={{ "aria-label": "search" }}
          onChange={handleFilter}
          data-testid="search"
        />
      </Search>
    </Toolbar>
  );
};

export default function EnhancedTable(props) {
  const { users, planets } = props;

  const [order, setOrder] = React.useState("asc");
  const [orderBy, setOrderBy] = React.useState("name");
  const [filter, setFilter] = React.useState("");
  const [selectedPlanet, setSelectedPlanet] = React.useState([]);
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleRequestSort = (_, property) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const handleClick = (_, name) => {
    console.log(name);
    const planet = planets.filter((planet) => {
      return planet.planet === name;
    });
    setSelectedPlanet(planet[0]);
    handleClickOpen();
  };

  const filteredUserList = () => {
    if (users)
      return users.filter((user) => {
        return user.name?.indexOf(filter) !== -1;
      });
  };

  const handleFilter = (event) => {
    console.log(event.target.value);
    if (event.target.value) {
      setFilter(event.target.value);
      return;
    }
    setFilter("");
  };

  const handleChangePage = (_, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  // Avoid a layout jump when reaching the last page with empty rows.
  const emptyRows =
    page > 0
      ? Math.max(0, (1 + page) * rowsPerPage - filteredUserList().length)
      : 0;

  return users ? (
    <div className="Box-Table" test>
      <Box sx={{ width: "100%" }}>
        <Paper sx={{ width: "100%", mb: 2 }}>
          <EnhancedTableToolbar handleFilter={handleFilter} />
          <TableContainer>
            <Table
              sx={{ minWidth: 750 }}
              aria-labelledby="tableTitle"
              size={"medium"}
            >
              <EnhancedTableHead
                order={order}
                orderBy={orderBy}
                onRequestSort={handleRequestSort}
                rowCount={filteredUserList().length}
              />
              <TableBody data-testid="table-data">
                {/* if you don't need to support IE11, you can replace the `stableSort` call with:
                 rows.slice().sort(getComparator(order, orderBy)) */}
                {stableSort(filteredUserList(), getComparator(order, orderBy))
                  .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                  .map((row) => {
                    return (
                      <TableRow
                        style={{
                          height: 33,
                        }}
                        hover
                        tabIndex={-1}
                        key={row.name}
                      >
                        <TableCell className="Table-Row">{row.name}</TableCell>
                        <TableCell className="Table-Row">
                          {row.height}
                        </TableCell>
                        <TableCell className="Table-Row">{row.mass}</TableCell>
                        <TableCell className="Table-Row">
                          {row.created}
                        </TableCell>
                        <TableCell className="Table-Row">
                          {row.edited}
                        </TableCell>
                        <TableCell className="Table-Row">
                          <div
                            className="Planet-Row"
                            onClick={(event) => handleClick(event, row.planet)}
                          >
                            {row.planet}
                          </div>
                        </TableCell>
                      </TableRow>
                    );
                  })}
                {emptyRows > 0 && (
                  <TableRow
                    style={{
                      height: 33,
                    }}
                  >
                    <TableCell colSpan={6} />
                  </TableRow>
                )}
              </TableBody>
            </Table>
          </TableContainer>
          <TablePagination
            rowsPerPageOptions={[5, 10, 25]}
            component="div"
            count={filteredUserList().length}
            rowsPerPage={rowsPerPage}
            page={page}
            onPageChange={handleChangePage}
            onRowsPerPageChange={handleChangeRowsPerPage}
          />
          <PlanetDialog
            open={open}
            handleClose={handleClose}
            selectedPlanet={selectedPlanet}
          />
        </Paper>
      </Box>
    </div>
  ) : null;
}
