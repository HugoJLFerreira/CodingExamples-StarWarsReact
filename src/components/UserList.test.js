import { render, screen, fireEvent } from "@testing-library/react";
import UserList from "./UserList";

// open, handleClose, selectedPlanet
const handleClose = jest.fn();
const users = [
  {
    name: "Luke Skywalker",
    height: "172",
    mass: "77",
    edited: "111",
    created: "111",
    planet: "test Planet",
  },
  {
    name: "C-3PO",
    height: "167",
    mass: "75",
    edited: "222",
    created: "111",
    planet: "test Planet",
  },
  {
    name: "R2-D2",
    height: "96",
    mass: "32",
    edited: "333",
    created: "555",
    planet: "test Planet",
  },
  {
    name: "Leia Organa",
    height: "150",
    mass: "49",
    edited: "55",
    created: "555",
    planet: "test Planet",
  },
  {
    name: "Obi-Wan Kenobi",
    height: "182",
    mass: "77",
    edited: "55",
    created: "55",
    planet: "test Planet",
  },
  {
    name: "Darth Vader",
    height: "202",
    mass: "136",
    edited: "55",
    created: "55",
    planet: "another test planet",
  },
  {
    name: "Owen Lars",
    height: "178",
    mass: "120",
    edited: "55",
    created: "55",
    planet: "another test planet",
  },
];

const planets = [
  {
    planet: "test Planet",
    climate: "temperate",
    diameter: "999000000",
    population: "1",
  },
  {
    planet: "another test planet",
    climate: "arid",
    diameter: "29000000",
    population: "1",
  },
];

test("renders learn react link", () => {
  render(<UserList users={users} planets={planets} />);
  const nameSorting = screen.getByText(/Name/i);
  fireEvent.click(nameSorting);
  const table = screen.getByTestId("table-data");
  expect(table).toMatchSnapshot();
  //Reorder by name
  fireEvent.click(nameSorting);
  const reverseOrderTable = screen.getByTestId("table-data");
  expect(reverseOrderTable).toMatchSnapshot();
  // expect(table).not.toEqual(reverseOrderTable);

  //Initial order
  fireEvent.click(nameSorting);
  const initialOrderTable = screen.getByTestId("table-data");
  expect(table).toEqual(initialOrderTable);

  //check r2-D2 is displayed and not filtered
  const r2 = screen.getByText(/R2-D2/i);
  expect(r2).toBeInTheDocument();

  //Search
  const search = screen.getByTestId("search").querySelector("input");
  fireEvent.click(search);
  fireEvent.change(search, { target: { value: "Luke" } });

  const luke = screen.getByText(/Luke Skywalker/i);
  expect(luke).toBeInTheDocument();

  //expect R2-D2 to be filtered out
  expect(r2).not.toBeInTheDocument();

  // Check Planet Dialog
  const planet = screen.getByText(/test Planet/i);
  fireEvent.click(planet);

  const diameter = screen.getByText(/999000000/i);
  expect(diameter).toBeInTheDocument();
});
