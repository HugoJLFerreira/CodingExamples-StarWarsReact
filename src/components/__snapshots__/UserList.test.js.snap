// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<tbody
  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  data-testid="table-data"
>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      R2-D2
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      96
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      32
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      555
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      333
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Owen Lars
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      178
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      120
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        another test planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Obi-Wan Kenobi
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      182
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      77
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Luke Skywalker
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      172
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      77
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Leia Organa
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      150
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      49
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      555
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Darth Vader
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      202
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      136
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        another test planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      C-3PO
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      167
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      75
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      222
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
</tbody>
`;

exports[`renders learn react link 2`] = `
<tbody
  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  data-testid="table-data"
>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      C-3PO
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      167
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      75
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      222
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Darth Vader
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      202
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      136
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        another test planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Leia Organa
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      150
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      49
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      555
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Luke Skywalker
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      172
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      77
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      111
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Obi-Wan Kenobi
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      182
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      77
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      Owen Lars
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      178
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      120
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      55
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        another test planet
      </div>
    </td>
  </tr>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
    style="height: 33px;"
    tabindex="-1"
  >
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      R2-D2
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      96
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      32
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      555
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      333
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium Table-Row css-1ex1afd-MuiTableCell-root"
    >
      <div
        class="Planet-Row"
      >
        test Planet
      </div>
    </td>
  </tr>
</tbody>
`;
